internal/services/network/network_manager_commit_resource.go

```diff
@@ -0,0 +1,348 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+	"github.com/tombuildsstuff/kermit/sdk/network/2022-07-01/network"
+)
+
+type ManagerCommitModel struct {
+	NetworkManagerId string   `tfschema:"network_manager_id"`
+	ScopeAccess      string   `tfschema:"scope_access"`
+	Location         string   `tfschema:"location"`
+	ConfigurationIds []string `tfschema:"configuration_ids"`
+	DeploymentStatus string   `tfschema:"deployment_status"`
+}
+
+type ManagerCommitResource struct{}
+
+func (r ManagerCommitResource) ResourceType() string {
+	return "azurerm_network_manager_commit"
+}
+
+func (r ManagerCommitResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.NetworkManagerCommitID
+}
+
+func (r ManagerCommitResource) ModelObject() interface{} {
+	return &ManagerCommitModel{}
+}
+
+func (r ManagerCommitResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"network_manager_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.NetworkManagerID,
+		},
+
+		"location": commonschema.Location(),
+
+		"scope_access": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(network.ConfigurationTypeConnectivity),
+				string(network.ConfigurationTypeSecurityAdmin),
+			}, false),
+		},
+
+		"configuration_ids": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"triggers": {
+			Type:     pluginsdk.TypeMap,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
```

Comment:
Thanks for clarifying. To be honest, as @tombuildsstuff mentioned this is somewhat of an anti-pattern with regards to Terraform, since updating constitutes replacing in all cases, and the effect of the replacement is asynchronous and out of band.

With this in mind, I'd be keen to minimize unexpected behavior and we should keep considering whether this resource is a good fit for Terraform. I think it could be, but I don't think we should include behavioral attributes like this and we should rather lean on existing ecosystem constructs like `replace_triggered_by`.

Given that this is all deployed asynchronously in the background anyway, and that Terraform will in almost all cases recreate the resource very quickly, I don't see this being a huge issue for most users? And if it is, then they should probably not be managing this resource with Terraform?

Reply:
This case is similar to existing [virtual_network_peering#triggers](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_network_peering#triggers) property. A replace means there will be a network downtime, while a simple update does not have this effect.

Reply:
OK, if it's going to involve real downtime in the case of rapidly "deleting" then recreating the deployment then we can leave in the `triggers` property, although for future development I'd note that we generally consider this an undesirable workaround for an incompatibly designed service, and that this is not a precedent (and neither is the same addition to the `virtual_network_peering` resource) for the same workaround to be applied elsewhere. We may also find that we need to remove this behavioral property in the next major version of the provider and, at that time, the service will need to be patched to support automatic redeployment.